### DISCUSSION TITLE
Adding repstr.pro to GDL

### DISCUSSION THREAD

**Discussion by Sandipan-Borthakur** [gnudatalanguage/gdl#1554] [Ideas]
Hii

I am Sandipan. I was using GDL to run a software which uses REPSTR function to replace a substring within a string. I do not think GDL has that by default, although I might be wrong( I am new to IDL and GDL). It would be nice to have it in GDL while installing. The code is already available [here](https://idlastro.gsfc.nasa.gov/ftp/pro/misc/repstr.pro). Let me know what you think.

**Comment by alaingdl** [comment]
Hi Sandipan

In fact this function is provided by the famous idlastron lib., which is packaged in Debian and usable on any operating system, as long as you add the path into your GDL path
see https://github.com/gnudatalanguage/gdl/wiki/Third-party-librairies

(then this external function will not be added inside the core of GDL)

you need to add that (the whole idlastro lib.) by your-self then add its path to GDL.

A simple way is to make : `git clone https://salsa.debian.org/debian-astro-team/idlastro`
(outside GDL). Then assuming you did that in a directory /my/path/to/idlastro then
after starting GDL you can do : `GDL> path_add, '/my/path/to/idlastro', /recur`
then : 
```
GDL> .r repstr
% Compiled module: REPSTR.
```

(A more classical way is to download from the NASA site, then to add the path into GDL_PATH outside GDL)


HTH

A.

PS: you can check you known/visible paths in GDL by the command `help,/path`, on my laptop I have

```GDL> help,/path
!PATH (Disabled, 17 directories)
/home/alaingdl/GDL/last-gdl/src/pro/ (147 files)
[...]
/home/alaingdl/IDLetGDL/Librairies/DebianAstroGIT/idlastro/pro/ (478 files)
[...]
```

**Reply by Sandipan-Borthakur** [reply]
Hii @alaingdl 

Thank you for the quick response. I just made a stupid mistake. I just noticed I didn't have the IDL Astronomy Library on my laptop although I added its path to GDL_PATH. I somehow deleted it and forgot to re-install. Bu thank you for the information and sorry about my stupidity!!!

**Comment by GillesDuvert** [comment]
@alaingdl 
Help,/PATH should not answer "(Disabled , XX directories) because it is Enabled and in Cache.
The good answer should be
`!PATH Cache (Enabled, XX directories)`
This is a one-liner to be done directly in the GDL GH repo

**Reply by Sandipan-Borthakur** [reply]
Since I have multiple directories I need for my software, I have gdlstartup.pro file which has the path to all the required directories. Before beginning to use GDL, I do the following - 

`export GDL_STARTUP=/fullpath/gdlstartup.pro` .

**Reply by GillesDuvert** [reply]
indeed.
paths should better be enclosed in quotes or double quotes `export GDL_STARTUP="/fullpath/gdlstartup.pro"`
works also with `IDL_STARTUP`, too.
